source/extensions/filters/network/thrift_proxy/conn_manager.cc

```diff
@@ -281,7 +286,6 @@ bool ConnectionManager::ActiveRpc::upstreamData(Buffer::Instance& buffer) {
 
 void ConnectionManager::ActiveRpc::resetDownstreamConnection() {
   parent_.read_callbacks_->connection().close(Network::ConnectionCloseType::NoFlush);
-  parent_.doDeferredRpcDestroy(*this);
```

Comment:
With this line removed, where does the cleanup happen instead?

Reply:
Because the downstream connection is closed, ConnectionManager::onEvent is invoked, which will destroy any items in the rpcs_ list. What was happening before was that the RPC was being destroyed but not removed from rpcs_.

Reply:
(Which causes a subsequent nullptr dereference.)

Reply:
Thanks, now it makes sense.